lumenlearning/lti-middleware
# DL3-48 add module ids and course paired params for deep linking response

## Description
Added the module_ids retrieved from the frontend to the request body and ensured that the course_paired request parameter is being populated from the database.

### Motivation and Context
These changes will allow for instructors to add only a subset of the links upon subsequent deep linking launches.

### Fixes
[DL3-48](https://lumenlearning.atlassian.net/browse/DL3-48)
Note: I have pulled the changes from [DL3-46](https://lumenlearning.atlassian.net/browse/DL3-46) into this branch as well since these tickets are interdependent.

## How Has This Been Tested?
1. Ensure DL3-46 is in place. Ensure that the LMS has the tool registered and that the org units have been added corresponding to a new test course.
2. Go to the new test course and open the deep linking menu.
3. Select a course for which there is data and module_ids (I've been using Lifespan Development).
4. Note that all of the checkboxes are selected by default and the button says "Add Course".
5. Click the button to add the course.
6. If DEBUG logging is turned on in AWS, then the logs will indicate that a module_ids value of `null` was sent in the body and that a request parameter of coursePaired=false was included in the request URL to Valkyrie.
7. Confirm that all of the links are added to the course for this case (for Lifespan Development, this is 40 links).
8. Click on one of the links in the course (I've been using a Study Plan link) to sync the lineitems. (NOTE: Lineitems must be synced for subsequent Deep Links to be added.)
9. If desired, create a new module to differentiate the previous steps from the next steps.
10. In the new module, open the deep linking menu.
11. Select the same course as previously (pending DL3-45 being merged in which case the course will be remembered).
12. Note that all of the checkboxes are unchecked by default and that the button says "Add Content".
13. Select the Select All checkbox and click the Add Content button.
14. Note that all of the content except the instructor resources are added (for Lifespan Development, this is 29 links).
15. Repeat steps 9-12.
16. Select only one or two modules and click the Add Content button.
17. Note that only the links for the selected modules are added (NOTE: A bug currently exists in DL3-49 which actually adds all modules EXCEPT the selected modules. Once this bug has been fixed, this step will occur as expected.)

## Screenshots
N/A

---

## Database changes
N/A

## ⚠️ Deployment instructions ⚠️
N/A

## New ENV variables
N/A

---

## Checklist
